### PR TITLE
Reject annotations on ivars defined in base class

### PR DESCRIPTION
**Pull request by waj**
Following up from #9499 

This raises an error instead of silently ignoring annotations on instance variables defined in a base class.

### CODE REVIEW

File: spec/compiler/semantic/annotation_spec.cr [hunk 1]
```diff
@@ -918,6 +918,23 @@ describe "Semantic: annotation" do
...
+      class Base
+        @x : Nil
+      end
+
+      class Child < Base
+        @[Foo]
+        @x : Nil
+      end
+      ),
+      "can't annotate an instance variable not defined in this class"
```
Comment (bcardiff) [inline spec/compiler/semantic/annotation_spec.cr:935]: Is the message right? Shoudn't it be: `can't annotate an instance variable defined in this class` ?
(as in without the _not_?)

Comment (jhass) [reply spec/compiler/semantic/annotation_spec.cr:935]: The point is that it's defined in `Base` and only *re*defined in `Child` and that's exactly what we want to forbid annotations on. Annotating it in `Base` is ok and annotating it in `Child` if *not* defined in `Base` is ok.

Comment (jhass) [reply spec/compiler/semantic/annotation_spec.cr:935]: But yeah, the double negation  might be a bit hard to follow.  Maybe

>  can only annotate instance variables defined in this class

or even

> cannot annotate @x here because it was first defined in Base

Comment (waj) [reply spec/compiler/semantic/annotation_spec.cr]: Done. More detailed and precise error is always better. Thanks!

File: src/compiler/crystal/semantic/type_declaration_processor.cr [hunk 2]
```diff
@@ -269,7 +269,7 @@ struct Crystal::TypeDeclarationProcessor
 
       # Reject annotations to existing instance var
       type_decl.annotations.try &.each do |_, ann|
-        ann.raise "can't annotate an instance variable not defined in this class"
+        ann.raise "can't annotate #{name} here because it was first defined in #{supervar.owner}"
```
Comment (Sija) [inline src/compiler/crystal/semantic/type_declaration_processor.cr:272]:
```suggestion
        ann.raise "can't annotate #{name} in #{owner} because it was first defined in #{supervar.owner}"
```

Comment (waj) [reply src/compiler/crystal/semantic/type_declaration_processor.cr:272]: Sounds good. I'll add the commit myself to fix the spec.
